liualgotrader/data/alpaca.py

```diff
@@ -32,12 +32,52 @@ def __init__(self):
             raise AssertionError(
                 "Failed to authenticate Alpaca RESTful client"
             )
+        # for requesting market snapshots by chunk of symbols
+        self.symbol_chunk_size = 1000
 
-    def get_symbols(self) -> List[Dict]:
+    def get_symbols(self) -> List[str]:
         if not self.alpaca_rest_client:
             raise AssertionError("Must call w/ authenticated Alpaca client")
 
-        return self.alpaca_rest_client.list_assets()
+        symbols = [
+            asset.symbol for asset in self.alpaca_rest_client.list_assets(
+                status='active', asset_class='us_equity'
+            ) if asset.tradable
+        ]
+        return symbols
+
+    def get_market_snapshot(self) -> List[Dict]:
+        # parse market snapshots per chunk of symbols
+        symbols = self.get_symbols()
+        raw_snapshots_dict = {}
+        for i in range(0, len(symbols), self.symbol_chunk_size):
```

Comment:
@riven314 why not use concurrency to pull chunks in parallel and combine?

Reply:
now I used ThreadPoolExecutor to achieve concurrency. Speed wise it takes ~9 sec (v.s. ~15 sec in the past)